### ISSUE TITLE
warn 'style-warning with arguments

### ISSUE THREAD

**Issue by lispm** [shop-planner/shop3#20]
In method process-op in shop3/io/input.lisp :

```
(warn 'style-warning :format-control "Operator ~a has implicit conjunction in preconditions.~%This is deprecated."
                               :format-arguments (list task))
```

The condition STYLE-WARNING has not format control and no format arguments. Thus the WARN function is being called with the wrong arguments. Some implementations will signal an error then.

ECL:

```
> (warn 'style-warning :format-control "foo" :format-arguments '())

Condition of type: SIMPLE-PROGRAM-ERROR
Unknown initialization option :FORMAT-ARGUMENTS for class #<The STANDARD-CLASS STYLE-WARNING>

```

**Comment by rpgoldman** [comment]
Thanks!  I think at least some lisp implementations extend `style-warning` to accept these arguments.

Just checked, and SBCL and Allegro accept them and Clozure CL rejects them.  I'm inclined to shadow `style-warning` to add those initargs and an appropriate report function, to cover over the incompatibility.

**Comment by rpgoldman** [comment]
Ah. Looking a little further, I think on SBCL and Allegro, `SHOP3.COMMON:STYLE-WARNING` will just be the same as `CL:STYLE-WARNING` but on ECL and CCL, it will be a subclass of both `STYLE-WARNING` and `SIMPLE-WARNING`.  Does that seem reasonable to you?  Do you know if Lispworks accepts the `SIMPLE-WARNING` initargs to `STYLE-WARNING`?

**Comment by lispm** [comment]
ASDF does this:

```
(define-condition simple-style-warning
      #+sbcl (sb-int:simple-style-warning) #-sbcl (simple-condition style-warning)
    ())
```

Looks okay to me.

> Do you know if Lispworks accepts the SIMPLE-WARNING initargs to STYLE-WARNING?

I have seen it fail in: CLISP, ECL, LispWorks, CCL.

**Comment by rpgoldman** [comment]
Looking at this, I see that it should be a `shop-condition` anyway. So I will add a `shop-domain-warning` class with the right inheritance pattern.

**Comment by rpgoldman** [comment]
Fixed in 4f21df281d3390340a66a4a486d8f363cb57160f
